spec/consensus/proposer-based-timestamp/pbts-algorithm_002_draft.md

```diff
@@ -0,0 +1,134 @@
+A value that is re-proposed retains its original proposal time, assigned by its original proposer.
+In other words, once assigned, the proposal time of a value is definitive.
+
+In the [first version][v1] of this specification, proposals were defined as pairs `(v, time)`.
+In addition, the same value could be proposed, in different rounds, associated to distinct times.
+Since this possibility does not exist in this second specification, the proposal time became part of the proposed value.
+With this simplification, several small changes to the [arXiv][arXiv] algorithm, replacing `v` by `(v, t)`, are not longer required.
+
+## Time Monotonicity
+
+Values decided in successive heights of consensus must have increasing times, so:
+
+- Monotonicity: for any process `p` and any two decided heights `h` and `h'`, if `h > h'` then `decision_p[h].time > decision_p[h'].time`.
+
+For ensuring time monotonicity, it is enough to ensure that a value `v` proposed by process `p` at height `h_p` has `v.time > decision_p[h_p-1].time`.
+So, if process `p` is the proposer of a round of height `h_p` and reads from its clock a time `now_p <= decision_p[h_p-1]`,
+it should postpone the generation of its proposal until `now_p > decision_p[h_p-1]`.
+
+Notice that monotonicity is not introduced by this proposal, being already ensured by [`bfttime`][bfttime].
+In `bfttime`, the `Timestamp` field of every `Precommit` message of height `h_p` sent by a correct process is required to be larger than `decision_p[h_p-1].time`, as one of such `Timestamp` fields becomes the time assigned to a value proposed at height `h_p`.
+
+The time monotonicity of values proposed in heights of consensus is verified by the `valid()` predicate, to which every proposed value is submitted.
+A value rejected by the `valid()` implementation is not accepted by any correct process.
+
+## Timely Proposals
+
+PBTS introduces a new requirement for a process to accept a proposal: the proposal must be `timely`.
+It is a temporal requirement, associated to a couple of synchronous assumptions regarding the behavior of processes and the network.
+
+The evaluation of the `timely` requirement requires comparing the proposal's sending time with the proposal's receiving time.
+As these two time values can be read from different clocks, at different processes, we need to assume that processes' clocks are synchronized.
+As these two times refer to two distinct events, we need to assume a minimum and a maximum real time interval between the occurrence of the two events.
+
+The two synchronous assumptions adopted to evaluate the `timely` predicate are:
+- Synchronized clocks: the values read from clocks of any two correct processes at the same instant of real time differ by at most `PRECISION`;
+- Bounded transmission delays: the real time interval between the sending of a proposal at a correct process, and the reception of the proposal at any correct process is upper bounded by `MSGDELAY`.
```

Comment:
I guess msgdelay is defined as in [[PBTS-MSG-D.0]](https://github.com/tendermint/spec/blob/master/spec/consensus/proposer-based-timestamp/pbts-sysmodel_001_draft.md#pbts-msg-d0), that is, in terms of clock time?

Reply:
Yes, but we assume that processes clocks progress about at the same rate as the real time, when considering short intervals?

Actually, does assuming this with respect of clock time bring us any advantage?

Reply:
It was just easier to write up as you just need one definition, rather than 1) Msg delay relative to real-time 2) clocks relative to real-time.